python/segyio/trace.py

```diff
@@ -201,7 +201,8 @@ def __getitem__(self, i):
         try:
             i = self.wrapindex(i)
             buf = np.zeros(n_elements, dtype = self.dtype)
-            return self.filehandle.gettr(buf, i, 1, 1, start, stop, step, n_elements)
+            tr = self.filehandle.gettr(buf, i, 1, 1, start, stop, step, n_elements)
+            return tr if n_elements > 1 else tr[0]
```

Comment:
Does this also make slices of size 1 return a scalar? I.e. is `f.trace[5, 7:8]` identical to `f.trace[5, 7]`? I didn't test it but I think they should be different; the first one should return an array of size 1 (as it already does on master).